data/autoyast_sle12/create_hdd/create_hdd_sles.xml.ep

```diff
@@ -0,0 +1,410 @@
+      <FW_ALLOW_PING_FW>yes</FW_ALLOW_PING_FW>
+      <FW_DEV_EXT>eth0</FW_DEV_EXT>
+      <FW_SERVICES_ACCEPT_EXT>0/0,tcp,22</FW_SERVICES_ACCEPT_EXT>
+    </firewall>
+    % }
+    <report>
+        <errors>
+            <log config:type="boolean">true</log>
+            <show config:type="boolean">true</show>
+            <timeout config:type="integer">0</timeout>
+        </errors>
+        <messages>
+            <log config:type="boolean">true</log>
+            <show config:type="boolean">true</show>
+            <timeout config:type="integer">0</timeout>
+        </messages>
+        <warnings>
+            <log config:type="boolean">true</log>
+            <show config:type="boolean">true</show>
+            <timeout config:type="integer">0</timeout>
+        </warnings>
+        <yesno_messages>
+            <log config:type="boolean">true</log>
+            <show config:type="boolean">true</show>
+            <timeout config:type="integer">0</timeout>
+        </yesno_messages>
+    </report>
+    <keyboard>
+        <keyboard_values>
+            <delay/>
+            <discaps config:type="boolean">false</discaps>
+            <numlock>bios</numlock>
+            <rate/>
+        </keyboard_values>
+        <keymap>english-us</keymap>
+    </keyboard>
+    <language>
+        <language>en_US</language>
+        <languages/>
+    </language>
+    <ntp-client>
+        <ntp_policy>auto</ntp_policy>
+    </ntp-client>
+    <software>
+        <products config:type="list">
+            <product><%= uc $get_var->('SLE_PRODUCT') %></product>
+        </products>
+        <packages config:type="list">
+            <!-- Workaround for bsc#1202234: [addon]-release packages are missing after autoyast installation. -->
+            % foreach (values %$addons) {
+            <package><%= lc($_->{name}) %>-release</package>
+            % }
+            <!-- end of workaround -->
+        </packages>
+	% if ($check_var->('PATTERNS', 'all')) {
+	<patterns config:type="list">
+	  <pattern>Basis-Devel</pattern>
+          <pattern>Minimal</pattern>
+          <pattern>SDK-C-C++</pattern>
+          <pattern>SDK-Doc</pattern>
+          <pattern>SDK-YaST</pattern>
+          <pattern>WBEM</pattern>
+          <pattern>apparmor</pattern>
+          <pattern>base</pattern>
+          <pattern>dhcp_dns_server</pattern>
+          <pattern>directory_server</pattern>
+	  <pattern>documentation</pattern>
+	  <pattern>file_server</pattern>
+	  <pattern>fips</pattern>
+          <pattern>gateway_server</pattern>
+          <pattern>gnome-basic</pattern>
+          <pattern>kvm_server</pattern>
+	  <pattern>kvm_tools</pattern>
+          <pattern>lamp_server</pattern>
+          <pattern>mail_server</pattern>
+          <pattern>ofed</pattern>
+          <pattern>printing</pattern>
+          <pattern>smt</pattern>
+          <pattern>x11</pattern>
+	  <pattern>yast2</pattern>
+        </patterns>
+	% }
+	% unless ($check_var->('PATTERNS', 'all')) {
```

Comment:
We probably don't want gnome-basic and X11 if we install a base server. I would just put no `<pattern>` section at all, as in create_hdd_ha_sles : then the default patterns for the selected product will be selected.
OR, if you want to be able to select specific patterns you could try to use this section from qam's sle 12 profile:  https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/a395a6d9103cb2f20c112750e4ab0935c25cf393/data/autoyast_qam/12_installation.xml.ep#L322.

It looks like the only difference between this file and create_hdd_ha_sles is this patterns section (?) so I would merge the two files.

Reply:
Thanks. We need generate default/all patterns qcow, so we need the 'patterns' section and the condition. Maybe I can update the profile name to make it better to understand for regression migration test.

Reply:
Hi, I understand we need the patterns if PATTERNS=all, but I don't nderstand why you want to install gnome and x11 on a text-based server. Then I see two problems:

- if we have PATTERNS variable empty, then we should  remove this section to let the system automatically detect the right patterns:
```
% unless ($check_var->('PATTERNS', 'all')) {
        <patterns config:type="list">
          <pattern>Minimal</pattern>
          <pattern>apparmor</pattern>
          <pattern>base</pattern>
          <pattern>documentation</pattern>
          <pattern>gnome-basic</pattern>
          <pattern>x11</pattern>
        </patterns>
	% }
```

- if we plan to do something different like "PATTERNS=base,mail_server", then this section is wrong.

In such case we better do this:
```
% if  (($get_var->(PATTERNS) and  !($check_var->('PATTERNS', 'all')) {
    <patterns config:type="list">
      % for my $pattern (@$patterns) {
      <pattern><%= $pattern %></pattern>
      % }
    </patterns>
% }
```
Or, again, we could use a normal profile that we will use only for one test suite, with no scripting/conditions at all.

Reply:
We have two kinds of PATTERNS (all/default), and this ticket aims to generate qcow to replace SLES-12-SP5-aarch64-GM-SDK-allpatterns.qcow2 which is gnome and not textmode, to keep the migration test so we need generate gnome qcow.
I updated the profile for PATTERNS=default to avoid misunderstanding, and so that it will get patterns automatically if no PATTERNS set. In fact, we only support PATTERNS=all or PATTERNS=default, anyway, to make the profile more strict is better. Thanks.

Reply:
Ah ok, now I understand. Looks good like this.